ambry-clustermap/src/main/java/com.github.ambry.clustermap/AmbryPartition.java

```diff
@@ -58,6 +60,21 @@
 
   @Override
   public PartitionState getPartitionState() {
+    // If there was a change to the sealed state of replicas in the cluster manager since the last check, refresh the
```

Comment:
Following your comment here, since partition states are very rarely changed, does it makes sense to add a callback to "HelixClusterManager" which "AmbryPartition" will call into whenever there is a state change. So, getWritablePartitions() doesn't need to go over 1000s of partitions everytime, considering the freq with which we make calls to getWritablePartitions().